src/client/dfuse/dfuse.h

```diff
@@ -623,9 +617,11 @@ struct dfuse_inode_entry {
 	 */
 	d_list_t                 ie_htl;
 
-	/* Time of last kernel cache update.
-	 */
-	struct timespec          ie_cache_last_update;
+	/* Time of last kernel cache metadata update */
+	struct timespec          ie_mcache_last_update;
+
+	/* Time of last kernel cache data update, also used for kernel readdir caching. */
+	struct timespec          ie_dcache_last_update;
```

Comment:
Because the kernel caches readdir in the page cache where data would go.  This allows it to be seperate from the directory metadata which would be the mtime, ownership etc.